pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseCombineOperator.java

```diff
@@ -118,6 +119,8 @@ public void runJob() {
     IntermediateResultsBlock mergedBlock;
     try {
       mergedBlock = mergeResults();
+    } catch (InterruptedException ie) {
+      throw new QueryCancelledException("Cancelled while merging results blocks", ie);
```

Comment:
(minor) Consider not including the `InterruptedException` for cleaner response

Reply:
The `e.getMessage()` is called to compose an info log so the stack trace won't be printed out. The cancel resp msg is composed separately w/o the stack trace at all. I was adding a DEBUG log level to print the full stack trace as needed but missed in the PR. Will add that DEBUG log back.